src/item.cpp

```diff
@@ -3998,20 +3992,33 @@ int item::gun_pierce( bool with_ammo ) const
     return ret;
 }
 
-int item::gun_recoil( bool with_ammo ) const
+int item::gun_handling() const
 {
     if( !is_gun() ) {
         return 0;
+
     }
-    int ret = type->gun->recoil;
-    if( with_ammo && ammo_data() ) {
-        ret += ammo_data()->ammo->recoil;
-    }
+    double handling = type->gun->handling;
     for( const auto mod : gunmods() ) {
-        ret += mod->type->gunmod->recoil;
+        handling += mod->type->gunmod->handling;
     }
-    ret += damage() * 15;
-    return ret;
+
+    return std::max( pow( ( weight() / 250 ), 0.8 ) * pow( handling / 10.0, 1.2 ), 0.0 );
```

Comment:
Handling != accuracy. Handling defines how well recoil can be controlled. For single shots it has no effect on accuracy. It does of course affect how useful bursts are. Repeated semi-auto shots of a powerful round out of a small gun could also lead to accumulation of recoil. Some of the variable naming in the project is unfortunate. We might consider renaming `MIN_RECOIL` to `AIM_LEVEL` or an equivalent.

Recoil is definitely an application of newtonian physics with weight one of the two significant factors. We have mods such as lightweight furniture that are correctly handled via this. There's a lot less room for trouble if the algorithms mirror reality as closely as possible. The folding stock mod I wrote is a good example of where that was done wrong - it has produced numerous special cases which I now need to go through and fix.

The powers basically just bias the algorithm against `weight` and in favour of `handling`. It's worth noting that you can `--dump-stats`. I'm not averse to adjusting the algorithm but I'd like to avoid having to adjust the JSON values too much as they are scaled against IRL values (kinetic energy) and that's a useful feature of the implementation when we come to adding new items.